types/index.d.ts

```diff
@@ -130,9 +130,8 @@ declare namespace Realm {
         clientResetAfter?: ClientResetAfterCallback;
     }
 
-    interface BaseSyncConfiguration<ClientResetModeT = ClientResetMode>{
+    interface BaseSyncConfiguration{
```

Comment:
`comment` I guess the idea with the generic was to avoid duplicating the `clientReset` key but for whatever reason I didn't end up going through with it 🤷‍♂️ happy to remove for now

Reply:
I just noticed it wasn't being used so I thought I'd remove it.  Not really related to this PR, but also nothing ground breaking